src/cli/commands/joinNetwork.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ArweaveSigner, mIOToken } from '../../node/index.js';
+import { JoinNetworkCLIOptions } from '../types.js';
+import {
+  assertEnoughBalance,
+  confirmationPrompt,
+  formatIOWithCommas,
+  gatewaySettingsFromOptions,
+  jwkToAddress,
+  requiredJwkFromOptions,
+  requiredOperatorStakeFromOptions,
+  writeActionTagsFromOptions,
+  writeIOFromOptions,
+} from '../utils.js';
+
+export async function joinNetwork(options: JoinNetworkCLIOptions) {
+  const jwk = requiredJwkFromOptions(options);
+  const address = jwkToAddress(jwk);
+  const io = writeIOFromOptions(options, new ArweaveSigner(jwk));
+
+  const ioQuantity = requiredOperatorStakeFromOptions(options);
+  const mIOOperatorStake = ioQuantity.toMIO().valueOf();
+
```

Comment:
nit: could get more typesafe and concise with `requiredIOFromOptions(options, OptionKeysENUM.operatorStake)`  `requiredStringFromOptions(options, OptionKeysENUM.operatorStake)`  `requiredArrayFromOptions(options, OptionKeysENUM.operatorStake)`  pattern. could get further shared with just `requiredOption(options, optionMap[value])` and in optionMap we define `type: IO` and have a switch for string, number, array, IO. Or even furthermore, a parseOption method could run on each provided option on the global runCommand -- with added AR/ETH address validation for very fast fails.